include/zephyr/drivers/flash/stm32_flash_api_extensions.h

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2023 Google Inc
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/flash.h>
+
+enum stm32_ex_ops {
+	/*
+	 * STM32 sector write protection control.
+	 *
+	 * As an input this operation takes a structure with two sector masks,
+	 * first mask is used to enable protection on sectors, while second mask
+	 * is used to do the opposite. If both masks are 0, then protection will
+	 * remain unchanged. If same sector is set on both masks, protection
+	 * will be enabled.
+	 *
+	 * As an output, sector mask with enabled protection is returned.
+	 * Input can be NULL if we only want to get protected sectors.
+	 * Output can be NULL if not needed.
+	 */
+	FLASH_STM32_EX_OP_SECTOR_WP = FLASH_EX_OP_VENDOR_BASE,
```

Comment:
Just a note, no change required: the header may provide `flash_stm32_ex_op_sector_wp_get()` `flash_stm32_ex_op_sector_wp_set()` inline functions in this header that will be passing reduced set of parameters to flash_ex_op, users may appreciate. 
This would probably work for all the others `FLASH_STM32_EX` commands provided.